tools/Nethermind.Tools.Kute/MessageProvider/JsonRpcMessageProvider.cs

```diff
@@ -26,26 +22,12 @@ await foreach (var msg in _provider.Messages)
         {
             var jsonDoc = JsonSerializer.Deserialize<JsonDocument>(msg);
 
-            switch (jsonDoc?.RootElement.ValueKind)
+            yield return jsonDoc?.RootElement.ValueKind switch
             {
-                case JsonValueKind.Object:
-                    yield return new JsonRpc.SingleJsonRpc(jsonDoc);
-                    break;
-                case JsonValueKind.Array:
-                    if (_unwrapBatches)
-                    {
-                        foreach (JsonElement single in jsonDoc.RootElement.EnumerateArray())
-                        {
-                            yield return new JsonRpc.SingleJsonRpc(JsonDocument.Parse(single.ToString()));
-                        }
-                    }
-                    yield return new JsonRpc.BatchJsonRpc(jsonDoc);
```

Comment:
A double enumeration happens here: if we unwrap batches we yield the batch elements, but we also yield the batch request at the end (most likely a missing `else` branch)

Reply:
@kamilchodola ?